sed/loader/flash/buffer_handler.py

```diff
@@ -79,160 +129,137 @@ def _schema_check(self) -> None:
                 )
 
     def _fill_dataframes(self):
         """
         Reads all parquet files into one dataframe using dask and fills NaN values.
         """
-        dataframe = dd.read_parquet(self.buffer_paths, calculate_divisions=True)
-        file_metadata = get_parquet_metadata(
-            self.buffer_paths,
-            time_stamp_col=self._config.get("time_stamp_alias", "timeStamp"),
-        )
-        self.metadata["file_statistics"] = file_metadata
-
-        fill_channels: list[str] = get_channels(
-            self._config["channels"],
-            ["per_pulse", "per_train"],
-            extend_aux=True,
-        )
-        index: list[str] = get_channels(index=True)
-        overlap = min(file["num_rows"] for file in file_metadata.values())
-
-        dataframe = forward_fill_lazy(
-            df=dataframe,
-            columns=fill_channels,
-            before=overlap,
-            iterations=self._config.get("forward_fill_iterations", 2),
-        )
-        self.metadata["forward_fill"] = {
-            "columns": fill_channels,
-            "overlap": overlap,
-            "iterations": self._config.get("forward_fill_iterations", 2),
-        }
-
-        # Drop rows with nan values in electron channels
-        df_electron = dataframe.dropna(
-            subset=get_channels(self._config["channels"], ["per_electron"]),
-        )
-
-        # Set the dtypes of the channels here as there should be no null values
-        channel_dtypes = get_channels(self._config["channels"], "all")
-        config_channels = self._config["channels"]
-        dtypes = {
-            channel: config_channels[channel].get("dtype")
-            for channel in channel_dtypes
-            if config_channels[channel].get("dtype") is not None
-        }
+        # Loop over the electron and timed dataframes
+        file_stats = {}
+        filling = {}
+        for typ in DF_TYP:
+            # Read the parquet files into a dask dataframe
+            df = dd.read_parquet(self.fp[typ], calculate_divisions=True)
+            # Get the metadata from the parquet files
+            file_stats[typ] = get_parquet_metadata(
+                self.fp[typ],
+                time_stamp_col=self._config.get("time_stamp_alias", "timeStamp"),
+            )
 
-        # Correct the 3-bit shift which encodes the detector ID in the 8s time
-        if self._config.get("split_sector_id_from_dld_time", False):
-            df_electron, meta = split_dld_time_from_sector_id(
-                df_electron,
-                config=self._config,
+            # Forward fill the non-electron channels across files
+            overlap = min(file["num_rows"] for file in file_stats[typ].values())
+            iterations = self._config.get("forward_fill_iterations", 2)
+            df = forward_fill_lazy(
+                df=df,
+                columns=self.fill_channels,
+                before=overlap,
+                iterations=iterations,
```

Comment:
You still do this here again?

Reply:
This was for filling between different files. The previous operation is filling within files. 

I will take a look again.

Reply:
You are right.